gr-utils/python/modtool/modtool_add.py

```diff
@@ -47,43 +50,25 @@ def __init__(self):
         self._skip_cmakefiles = False
         self._license_file = None
 
-    @staticmethod
-    def setup_parser(parser):
-        parser.add_argument("-t", "--block-type", choices=ModToolAdd._block_types,
-                help="One of %s." % ', '.join(ModToolAdd._block_types))
-        parser.add_argument("--license-file",
-                help="File containing the license header for every source code file.")
-        parser.add_argument("--copyright",
-                help="Name of the copyright holder (you or your company) MUST be a quoted string.")
-        parser.add_argument("--argument-list",
-                help="The argument list for the constructor and make functions.")
-        parser.add_argument("--add-python-qa", action="store_true", default=None,
-                help="If given, Python QA code is automatically added if possible.")
-        parser.add_argument("--add-cpp-qa", action="store_true", default=None,
-                help="If given, C++ QA code is automatically added if possible.")
-        parser.add_argument("--skip-cmakefiles", action="store_true",
-                help="If given, only source files are written, but CMakeLists.txt files are left unchanged.")
-        parser.add_argument("-l", "--lang", choices=('cpp', 'c++', 'python'),
-                help="Programing language")
-        ModTool.setup_parser_block(parser)
-        return parser
-
     def setup(self, options):
         ModTool.setup(self, options)
 
         self._info['blocktype'] = options.block_type
         if self._info['blocktype'] is None:
-            # Print(list out of blocktypes to user for reference)
-            print(str(self._block_types))
-            while self._info['blocktype'] not in self._block_types:
-                self._info['blocktype'] = input("Enter block type: ")
-                if self._info['blocktype'] not in self._block_types:
-                    print('Must be one of ' + str(self._block_types))
+            print (str(self._block_types))
+            with SequenceCompleter(sorted(self._block_types)):
+                while self._info['blocktype'] not in self._block_types:
+                    self._info['blocktype'] = input("Enter block type: ")
+                    if self._info['blocktype'] not in self._block_types:
+                        print ('Must be one of ' + str(self._block_types))
+
         # Allow user to specify language interactively if not set
         self._info['lang'] = options.lang
         if self._info['lang'] is None:
-            while self._info['lang'] not in ['c++', 'cpp', 'python']:
-                self._info['lang'] = input("Language (python/cpp): ")
+            language_candidates = ('cpp', 'python', 'c++')
+            with SequenceCompleter(language_candidates):
+                while self._info['lang'] not in language_candidates:
+                    self._info['lang'] = input("Language (python/cpp): ")
         if self._info['lang'] == 'c++':
```

Comment:
I assume that now that 'c++' is in 'language_candidates', these lines can be removed. Right?

Reply:
@primercuervo: I am not sure, how?

Reply:
No, forget it. I think I confused the purpose of these lines. There is no change needed here